### PR TITLE
[PULL REQUEST] Chemistry level

### PR DESCRIPTION
**Pull request by rgryan92**
Extending UCX mechanism to the top level of the model

### CODE REVIEW

**Comment by msulprizio** [comment]
@rgryan92 Did you run this update past the Chemistry Working Group and receive approval? I'm tagging Co-Chairs (@mje1398, @barronh, @luho0) so they are aware of this pull request.

On our [Model Development Priority page](http://wiki.seas.harvard.edu/geos-chem/index.php/GEOS-Chem_model_development_priorities#Feature_integration_not_yet_prioritized), we do have an item to "Extend UCX into the mesosphere" with Seb Eastham (MIT) and Eric Fleming (NASA/GMAO) listed as developers. @sdeastham Do you have any comments or concerns with this PR?

Also note in GEOS-Chem 13.2.1, we retired the tropchem simulation option so by default in gc_grid_mod.F90 we now have:
```
    ! Set maximum number of levels in the chemistry grid
    State_Grid%MaxChemLev  = State_Grid%MaxStratLev
```

In other words, `Input_Opt%LUCX` has been removed and all code previously associated with that logical switch is now the default. 

Would we want to reinstate an option for increasing the chemistry grid from the mesopause (i.e. `State_Grid%MaxStratLev`) to the top of the model (`State_Grid%NZ`) or would this be the default? @rgryan92 Did you run any timing tests to evaluate the impact on model performance when increasing the number of levels in the chemistry grid?

**Review by msulprizio — COMMENTED** [review]
@rgryan92 I have reviewed the changes in this pull request and have the following comments:

1. I would recommend removing `CATEGORY_ROCKET` from hcoi_gc_diagn_include.H since we do not have the emissions updates for this new category.

2. @yantosca has greatly modified and cleaned up gckpp_HetRates.F90 via PR https://github.com/geoschem/geos-chem/pull/827. Will the update to "Add heterogeneous reaction of CLNO3+HCl-->HNO3+Cl2" conflict with those changes (slated for 13.3.0)? Is that update needed for increasing the chemistry level or is it a separate science update and/or bug fix?

**Comment by eamarais** [comment]
Hi @msulprizio, the extension of the model chemistry to the top model layer doesn't actually work. @rgryan92 and I discovered a few weeks after making this change that ozone is very rapidly depleted in the top model layers. We suspect it's an issue with a switch in the model (like being hardcoded to use linear chemistry) that we weren't able to find and resolve. I think @sdeastham is working with Peter Fleming to update the model to use UCX chemistry throughout.

**Comment by msulprizio** [comment]
> Hi @msulprizio, the extension of the model chemistry to the top model layer doesn't actually work. @rgryan92 and I discovered a few weeks after making this change that ozone is very rapidly depleted in the top model layers. We suspect it's an issue with a switch in the model (like being hardcoded to use linear chemistry) that we weren't able to find and resolve. I think @sdeastham is working with Eric Fleming to update the model to use UCX chemistry throughout.

Thanks @eamarais! In that case, I will close this PR and we will wait for updates from @sdeastham and Eric Fleming.
